### PR TITLE
per-config display of unit selections

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request provides the infrastructure for configs (and downstream apps) to dictate which display units should be shown in the plugin, as well as including a time unit (currently unused for all configs, but needed by lcviz).  The individual logic may need to be adjusted as unit conversion is enabled for the remaining configs (I just made a rough guess).  This PR also makes the surface-brightness API access read-only.

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Comment by kecnry** [comment]
ah, that was a missed update in the rebase, should be fixed now

File: jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.vue [hunk 1]
```diff
@@ -60,27 +73,29 @@
...
-        persistent-hint
-      ></v-select>
-    </v-row>
+      <v-row>
+        <v-select
+          :menu-props="{ left: true }"
+          attach
+          :items="spectral_y_type_items.map(i => i.label)"
+          v-model="spectral_y_type_selected"
+          :label="api_hints_enabled ? 'plg.spectral_y_type =' : 'Flux or Surface Brightness'"
```
Comment (rosteen) [inline jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.vue:87]: I think you missed this string in the rebase, unless you meant to change it back from the more general language.

Comment (kecnry) [reply jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.vue]: nope, missed that too 🤦   Reverted to "Spectral y-axis Type"

**Review by rosteen — APPROVED** [review]
Looks good to me now, thanks.

File: jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py [hunk 2]
```diff
@@ -97,30 +105,51 @@ def __init__(self, *args, **kwargs):
...
         self.flux_unit = UnitSelectPluginComponent(self,
                                                    items='flux_unit_items',
                                                    selected='flux_unit_selected')
 
+        self.has_angle = self.config in ('cubeviz', 'specviz', 'mosviz')
         self.angle_unit = UnitSelectPluginComponent(self,
                                                     items='angle_unit_items',
                                                     selected='angle_unit_selected')
 
+        self.has_sb = self.has_angle or self.config in ('imviz',)
```
Comment (pllim) [inline jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py:123]:
```suggestion
        self._has_sb = self.has_angle or self.config in ('imviz',)
```

As per "convention," should this be "hidden" then if read only?

**Review by pllim — APPROVED** [review]
Looks much cleaner code-wise. Thanks! Just a minor question, so approving. Merge at will.

File: jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py (hunk 2 above)
Comment (kecnry) [reply jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py:123]: Traitlets cannot have leading underscores, and this is not exposed to the user

**Pull request merged**